src/WebpackTools/middlewares/DevProxy.js

```diff
@@ -1,4 +1,38 @@
+/**
+ * Proxies all requests not served by Webpack in-memory bundling back
+ * to the underlying store.
+ *
+ * Tries to detect a case where the target URL is misconfigured and the
+ * store tries to redirect to http or https.
+ */
 const proxy = require('http-proxy-middleware');
+const { format, URL } = require('url');
+
+const RedirectCodes = [201, 301, 302, 307, 308];
+const findRedirect = message =>
+    RedirectCodes.includes(message.statusCode) && message.headers.location;
+
+const detectProtocolChange = (target, redirected) => {
```

Comment:
The name `throwOnProtocolChange` might be more descriptive of what this is doing.